src/ngx_srt_set_misc_module.c

```diff
@@ -0,0 +1,302 @@
+#include <stdint.h>
+#include <ngx_config.h>
+#include <ngx_core.h>
+
+#include <openssl/evp.h>
+#include <openssl/md5.h>
+
+#include "ngx_srt.h"
+
+
+typedef struct {
+    ngx_srt_complex_value_t  value;
+} ngx_srt_base64_ctx_t;
+
+typedef struct {
+    ngx_str_t                key;
+    ngx_str_t                iv;
+    ngx_srt_complex_value_t  value;
+} ngx_srt_encrypt_decrypt_ctx_t;
+
+enum {
+    ngx_encrypt_key_length = 256 / 8,
+    ngx_encrypt_iv_length = EVP_MAX_IV_LENGTH
+};
+
+
+static char *ngx_srt_set_misc_decrypt(ngx_conf_t *cf, ngx_command_t *cmd,
+    void *conf);
+static char *ngx_srt_set_misc_base64(ngx_conf_t *cf, ngx_command_t *cmd,
+    void *conf);
+
+
+static ngx_command_t  ngx_srt_set_misc_commands[] = {
+
+    { ngx_string("set_decode_base64"),
+      NGX_SRT_MAIN_CONF|NGX_CONF_TAKE2,
+      ngx_srt_set_misc_base64,
+      NGX_SRT_MAIN_CONF_OFFSET,
+      0,
+      NULL },
+
+    { ngx_string("set_aes_decrypt"),
+      NGX_SRT_MAIN_CONF|NGX_CONF_TAKE4,
+      ngx_srt_set_misc_decrypt,
+      NGX_SRT_MAIN_CONF_OFFSET,
+      0,
+      NULL },
+
+      ngx_null_command
+};
+
+static ngx_srt_module_t  ngx_srt_set_misc_module_ctx = {
+    NULL,                                  /* preconfiguration */
+    NULL,                                  /* postconfiguration */
+
+    NULL,                                  /* create main configuration */
+    NULL,                                  /* init main configuration */
+
+    NULL,                                  /* create server configuration */
+    NULL                                   /* merge server configuration */
+};
+
+ngx_module_t  ngx_srt_set_misc_module = {
+    NGX_MODULE_V1,
+    &ngx_srt_set_misc_module_ctx,             /* module context */
+    ngx_srt_set_misc_commands,                /* module directives */
+    NGX_SRT_MODULE,                        /* module type */
+    NULL,                                  /* init master */
+    NULL,                                  /* init module */
+    NULL,                                  /* init process */
+    NULL,                                  /* init thread */
+    NULL,                                  /* exit thread */
+    NULL,                                  /* exit process */
+    NULL,                                  /* exit master */
+    NGX_MODULE_V1_PADDING
+};
+
+
+static ngx_int_t
+ngx_srt_set_misc_base64_variable(ngx_srt_session_t *s, ngx_srt_variable_value_t *v,
+    uintptr_t data)
+{
+    ngx_str_t              val, decode_str;
+    ngx_srt_base64_ctx_t  *base64;
+
+    base64 = (ngx_srt_base64_ctx_t *) data;
+
+    ngx_log_debug0(NGX_LOG_DEBUG_STREAM, s->connection->log, 0,
+        "srt base64 started");
+
+    if (ngx_srt_complex_value(s, &base64->value, &val) != NGX_OK) {
+        ngx_log_error(NGX_LOG_ERR, s->connection->log, 0,
```

Comment:
change the severity to NOTICE (there will already be another error log if it fails)